### PR TITLE
File Name Convention Integration

### PR DESCRIPTION
**Pull request by collinss-jpl**
This PR adds integration of file name assignment for output products which may be tailored on a per-PGE basis to meet the file naming conventions expected for that PGE. Currently, this includes a base implementation within the base PGE post-processor class, as well as a tailored implementation for the existing DSWx-HLS PGE.

The file-renaming step occurs during post-processing during staging of all output products to the output location defined by the RunConfig. The Base PGE post-processor iterates over all products created by SAS execution, and renames each file based on the file's extension. During this step, the catalog metadata and PGE/SAS log is also written to disk using the appropriate file names. 

Inheritors of the Base PGE need only provide their own implementations for their specific renaming functions, leaving the rest of the logic to the implementation in the Base PGE.

### CODE REVIEW

File: src/opera/pge/base_pge.py [hunk 1]
```diff
@@ -247,39 +237,20 @@ def _run_sas_qa_executable(self):
...
-            msg = "Successfully created catalog metadata json file."
-            self.logger.info("pge_main", ErrorCode.CREATING_CATALOG_METADATA, msg)
-        else:
-            msg = f"SCHEMA ERROR: {met_file.get_error_msg()}"
-            self.logger.info("pge_main", ErrorCode.CREATING_CATALOG_METADATA, msg)
 
-    def _create_iso_metadata(self):
-        # TODO
-        pass
+        return MetFile(catalog_metadata)
```
Comment (JimHofman) [inline src/opera/pge/base_pge.py:251]: Very nice clean factoring of the code.

File: src/opera/pge/base_pge.py [hunk 2]
```diff
@@ -295,6 +266,196 @@ def _finalize_log(self):
...
+
+        The GeoTIFF filename for the Base PGE consists of:
+
+            <Core filename>_<inter_filename>.tif
+
+        Where <Core filename> is returned by PostProcessorMixin._core_filename()
+
+        Parameters
+        ----------
+        inter_filename : str, optional
```
Comment (JimHofman) [inline src/opera/pge/base_pge.py:312]: Is this really option.  Won't line 323 fail if this is not defined?

File: src/opera/pge/dswx_pge.py [hunk 3]
```diff
@@ -143,11 +231,8 @@ def run_postprocessor(self, **kwargs):
         print(f'Running postprocessor for {self._post_mixin_name}')
 
         self._run_sas_qa_executable()
-        self._create_catalog_metadata()
-        self._create_iso_metadata()
         self._validate_output()
         self._stage_output_files()
```
Comment (JimHofman) [inline src/opera/pge/dswx_pge.py:235]: Again nice factoring.

**Review by JimHofman — APPROVED** [review]

File: src/opera/pge/base_pge.py (hunk 2 above)
Comment (collinss-jpl) [reply src/opera/pge/base_pge.py:312]: Good find, I definitely wanted the interface to allow `inter_filename` to be optional, but this function, and the version in `dsxw_hls.py` should also be checking for `None` and handling appropriately.

**Pull request merged**